src/lib.rs

```diff
@@ -677,33 +677,27 @@ impl<T> SliceDeque<T> {
     /// ```
     #[inline]
     pub fn reserve_exact(&mut self, additional: usize) {
-        self.reserve(additional);
+        let old_len = self.len();
+        let new_cap = old_len.checked_add(additional).expect("overflow");
+        self.reserve_capacity(new_cap);
+        debug_assert!(self.capacity() >= old_len + additional);
     }
 
     /// Growth policy of the deque. The capacity is going to be a multiple of
-    /// the page-size anyways, so we just double on growth.
+    /// the page-size anyways, so we just double capacity when needed.
     #[inline]
-    fn grow_policy(&self) -> usize {
-        unsafe {
-            if intrinsics::unlikely(self.capacity() == 0) {
-                4
-            } else {
-                self.capacity() * 2
-            }
+    fn grow_policy(&self, additional: usize) -> usize {
+        let cur_cap = self.capacity();
+        let old_len = self.len();
+        let req_cap = old_len.checked_add(additional).expect("overflow");
+        if req_cap > cur_cap {
+            let dbl_cap = cur_cap.saturating_mul(2);
```

Comment:
While I wrote the comment above, I don't really follow its logic anymore. We can probably use a `1.5` (`3/4`)  growth factor here instead. I think it might be worth it to try that in a different PR.